python/neml2/pyzag/interface.py

```diff
@@ -146,20 +146,22 @@ def _setup_parameters(self, exclude_parameters):
         Args:
             exclude_parameters (list of str): NEML2 parameters to exclude
         """
+        self.parameter_names = []
         for pname, param in self.model.named_parameters().items():
             if "." in pname:
                 errmsg = "Parameter name {} contains a period, which is not allowed. \nMake sure Settings/parameter_name_separator='_' in the NEML2 input file."
                 raise ValueError(errmsg.format(pname))
             if pname in exclude_parameters:
                 continue
+            self.parameter_names.append(pname)
```

Comment:
I think it's better to add some comments explaining why we need an additional map between names, and also the use of `getattr` below. I bet we'll forget in the future.